docs/src/pages/demos/snackbars/ConsecutiveSnackbars.js

```diff
@@ -1,103 +1,114 @@
+        nextState.messageInfo = nextState.queue.shift();
+        nextState.open = true;
+      }
 
-    if (this.state.open) {
-      // immediately begin dismissing current message
-      // to start showing new one
-      this.setState({ open: false });
-    } else {
-      this.processQueue();
+      return nextState;
     }
-  };
+    case 'processQueue': {
+      if (state.queue.length === 0) return state;
 
-  processQueue = () => {
-    if (this.queue.length > 0) {
-      this.setState({
-        messageInfo: this.queue.shift(),
-        open: true,
-      });
+      const nextState = { ...state, open: true };
+      nextState.messageInfo = nextState.queue.shift();
+      return nextState;
+    }
+    case 'close': {
+      return { ...state, open: false };
     }
-  };
+    default: {
+      return state;
+    }
+  }
+}
+
+function ConsecutiveSnackbars() {
+  const classes = useStyles();
+  const [state, dispatch] = React.useReducer(reducer, DefaultState);
+
+  function handleClick(event, message) {
+    dispatch({ type: 'addItem', payload: message });
+  }
 
-  handleClose = (event, reason) => {
+  function processQueue() {
+    dispatch({ type: 'processQueue' });
+  }
+
+  function handleClose(event, reason) {
     if (reason === 'clickaway') {
       return;
     }
-    this.setState({ open: false });
-  };
 
-  handleExited = () => {
-    this.processQueue();
-  };
-
-  render() {
-    const { classes } = this.props;
-    const { messageInfo = {} } = this.state;
+    dispatch({ type: 'close' });
+  }
 
-    return (
-      <div>
-        <Button onClick={this.handleClick('Message A')}>Show message A</Button>
-        <Button onClick={this.handleClick('Message B')}>Show message B</Button>
-        <Snackbar
-          key={messageInfo.key}
-          anchorOrigin={{
-            vertical: 'bottom',
-            horizontal: 'left',
-          }}
-          open={this.state.open}
-          autoHideDuration={6000}
-          onClose={this.handleClose}
-          onExited={this.handleExited}
-          ContentProps={{
-            'aria-describedby': 'message-id',
-          }}
-          message={<span id="message-id">{messageInfo.message}</span>}
-          action={[
-            <Button key="undo" color="secondary" size="small" onClick={this.handleClose}>
-              UNDO
-            </Button>,
-            <IconButton
-              key="close"
-              aria-label="Close"
-              color="inherit"
-              className={classes.close}
-              onClick={this.handleClose}
-            >
-              <CloseIcon />
-            </IconButton>,
-          ]}
-        />
-      </div>
-    );
+  function handleExited() {
+    processQueue();
   }
-}
 
-ConsecutiveSnackbars.propTypes = {
-  classes: PropTypes.object.isRequired,
-};
+  return (
+    <div>
+      <Button onClick={e => handleClick(e, 'Message A')}>Show message A</Button>
+      <Button onClick={e => handleClick(e, 'Message B')}>Show message B</Button>
```

Comment:
Seems like we never need the event. Probably best to rename this createHandleClick and only ever accept a single message argument.